tests/CacheManagerTest.php

```diff
@@ -0,0 +1,361 @@
+      'wp_redirect', [
+        'return' => true
+      ]
+    );
+    \WP_Mock::userFunction(
+      'opcache_reset', [
+        'return' => true
+      ]
+    );
+    \WP_Mock::userFunction(
+      'wp_cache_flush', [
+        'return' => true
+      ]
+    );
+    \WP_Mock::userFunction(
+      'wp_redirect', [
+        'return' => true
+      ]
+    );
+
+    $mock->check_cache_request();
+  }
+
+  /**
+   * @covers \Niteo\WooCart\Defaults\CacheManager::__construct
+   * @covers \Niteo\WooCart\Defaults\CacheManager::redis_connect
+   */
+  public function testRedisConnect() {
+    $mock = \Mockery::mock( 'Niteo\WooCart\Defaults\CacheManager' )
+                      ->shouldAllowMockingProtectedMethods()
+                      ->makePartial();
+    $mock->shouldReceive( 'redis_connect' )
+         ->andReturn( true );
+
+    $this->assertTrue( $mock->redis_connect() );
+  }
+
+  /**
+   * @covers \Niteo\WooCart\Defaults\CacheManager::__construct
+   * @covers \Niteo\WooCart\Defaults\CacheManager::flush_cache
+   * @covers \Niteo\WooCart\Defaults\CacheManager::flush_opcache
+   * @covers \Niteo\WooCart\Defaults\CacheManager::flush_redis_cache
+   * @covers \Niteo\WooCart\Defaults\CacheManager::flush_fcgi_cache
+   */
+  public function testFlushCache() {
+    $method = self::getMethod( 'flush_cache' );
+    $mock   = \Mockery::mock( 'Niteo\WooCart\Defaults\CacheManager' )
+                      ->shouldAllowMockingProtectedMethods()
+                      ->makePartial();
+
+    $mock->shouldReceive( 'flush_opcache' )
+         ->andReturn( true );
+    $mock->shouldReceive( 'flush_redis_cache' )
+         ->andReturn( true );
+    $mock->shouldReceive( 'flush_fcgi_cache' )
+         ->andReturn( true );
+
+    $method->invokeArgs( $mock, [] );
+  }
+
+  /**
+   * @covers \Niteo\WooCart\Defaults\CacheManager::__construct
+   * @covers \Niteo\WooCart\Defaults\CacheManager::flush_fcgi_cache
+   */
+  // public function testFlushFcgiCache() {
+  //   $method = self::getMethod( 'flush_fcgi_cache' );
+  //   $plugins = new CacheManager();
+
+  //   $method->invokeArgs( $plugins, [] );
+  // }
+
+  /**
+   * @covers \Niteo\WooCart\Defaults\CacheManager::__construct
+   * @covers \Niteo\WooCart\Defaults\CacheManager::flush_redis_cache
+   */
+  public function testFlushRedisCache() {
+    $method = self::getMethod( 'flush_redis_cache' );
+    $mock   = \Mockery::mock( 'Niteo\WooCart\Defaults\CacheManager' )
+                      ->shouldAllowMockingProtectedMethods()
+                      ->makePartial();
+
+    \WP_Mock::userFunction(
+      'wp_cache_flush', [
+        'return' => true
+      ]
+    );
+
+    $mock->shouldReceive( 'redis_connect' )
+         ->andReturn( true );
+    $mock->connected = true;
+
+    $mock->redis = new stdClass();
+    $mock->redis->del = function( $key ) {
+      // Do nothing
+    };
+
+    \Mockery::mock( 'alias:\Predis\Collection\Iterator\Keyspace' );
+
+    $method->invokeArgs( $mock, [] );
```

Comment:
return a array that is used in for loop in code, and the above del mock will be able to assert that loop is working as expected